ons_alpha/private_media/utils.py

```diff
@@ -0,0 +1,131 @@
+import concurrent.futures
+import logging
+
+from collections.abc import Iterable
+from typing import NamedTuple
+
+from asgiref.local import Local
+from botocore.exceptions import ClientError
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models.fields.files import FieldFile
+from storages.backends.s3 import S3Storage
+from wagtail.models import Collection
+
+
+logger = logging.getLogger(__name__)
+
+
+_ROOT_COLLECTION_INFO = Local()
```

Comment:
Question: Given these collections are intended to be functionally immutable, any reason not to use an `lru_cache` here instead?

Reply:
I see what you mean. The main problem I was trying to solve was making these values available cheaply when creating or updating a lot of things in bulk, which this achieves. Despite the context, I can't help but feel reticent about caching things for longer. Let me think in this a bit more.